src/builder.rs

```diff
@@ -256,13 +263,11 @@ derivation {{
   allowSubstitutes = false;
   preferLocalBuild = true;
   system = builtins.currentSystem;
-  # this is to make nix rebuild for every test, I’m so sorry
-  random = {:?};
+  # this is to make nix rebuild for every test
+  random = builtins.currentTime;
```

Comment:
I’m not sure this is better than the tempdir, mainly because I’m afraid future versions of nix will stub that out to be always the same in pure mode (c.f. flakes)